src/journal.rs

```diff
@@ -0,0 +1,57 @@
+use std::fs::File;
+use std::path::Path;
+
+use sdjournal::journal::*;
+use sdjournal::iter::EntryIter;
+
+use crate::errors::*;
+
+#[derive(Debug)]
+pub struct SdJournal {
+    inner: Journal<File>,
+}
+
+impl SdJournal {
+    pub fn open_journal<P: AsRef<Path>>(path: P) -> Result<Self> {
```

Comment:
I think there is a bit of API mismatch here. To the best of my knowledge, the C API never works on a single file in isolation. I think the closest to this is `sd_journal_open_files`, which would take a slice of paths instead.

Perhaps we should start with an implementation of `sd_journal_open_files_fd` instead, and build the other ones of top of that?